comp/core/flare/flare.go

```diff
@@ -36,6 +37,7 @@ type dependencies struct {
 	InvAgent              inventoryagent.Component // TODO: (components) - Temporary dependencies until the status page is a Component and we don't need to call it in 'CompleteFlare'.
 	Params                Params
 	Providers             []types.FlareCallback `group:"flare"`
+	Wmeta                 workloadmeta.Component
```

Comment:
I think we don't want wlm as a dependency of the flare command. A solution would be to put it as a dependency of the main function

Reply:
Is there any side affect to make wmeta as a dependency here? One follow-up work of workloadmeta comp is to remove global state of workloadmeta store. That means we need to pass wmeta into pkgFlare.CompleteFlare which will require compFlare to acquire comp wmeta first.

Reply:
CompleteFlare will pass the wmeta comp to diagnose function which is necessary to fix `agent diagnose`.

Reply:
Basically the `flare` command should not really have this dependency. 

It normally registers some `FlareCallbacks` that should generate some files in the flare.

The CompleteFlare is an exception right now. It calls the diagnose command. We think the diagnose command should be the one that depends on workloadmeta.

Given that workloadmeta is still using a global state, we think it should be a dependency of the main function instead (because it's not a dependency of the flare).

Reply:
The final target is to not have any dependency like `workloadmeta` in the flare. Until the `diagnose` code is a component and we can remove its related code from `CompleteFlare` we can `workloadmeta` as a requirement here. 

We should make this dependency optional as the flare comp could be use in a configuration where `workloadmeta` is not required.

For this PR: I'd rather not have `workloadmeta` as a dependency for the flare until we actually use it. WDYT ?

Reply:
hi @hush-hush,

So where in the `agent flare` command code path, you think we could instantiate workloadmeta component? 
It is not clear to me what are our option here? could you help to define a possible solution ?

Reply:
If it is ok to have an wmeta as an optional dependency of flare, workloadmeta comp provides an OptionalModule() which can be determined by cliParams.local. 
But we still need to pass explicitly wmeta to diagnose because it is not converted to component yet.

Reply:
I updated pr to remove the dependency in flare. Instead, workloadmeta is loaded in main command functions(both flare and diagnose cli command). In the future, when diagnose is converted to component, we can move dependency into diagnose component.